packages/core/http/core-http-server/src/router/route.ts

```diff
@@ -160,7 +160,24 @@ export interface RouteConfigOptions<Method extends RouteMethod> {
     idleSocket?: number;
   };
 
-  /** A short, human-friendly description of this endpoint */
+  /**
+   * Human-friendly summary of this route, used for (OAS) documentation
+   *
+   * @example
+   * ```ts
+   * router.get({
+   *  path: '/api/foo/{id}',
+   *  access: 'public',
+   *  summary: `Retrieve foo resources given an ID`,
+   * })
+   * ```
+   */
+  summary?: string;
+
+  /**
+   * A description of this API. [CommonMark syntax](https://spec.commonmark.org)
+   * may be used for rich text
```

Comment:
What do you mean by 'may be used for rich text'? Is it that one can use rich text in the description or that the description might be used for rich text?

Reply:
in case it's helpful: https://spec.openapis.org/oas/v3.0.3#rich-text-formatting. I've also added an example to demonstrate per https://github.com/elastic/kibana/pull/184651#issuecomment-2150977763